src/MolSim.cpp

```diff
@@ -1,50 +1,136 @@
 
 #include "FileReader.h"
 #include "outputWriter/XYZWriter.h"
+#include "outputWriter/VTKWriter.h"
 #include "utils/ArrayUtils.h"
 
 #include <iostream>
 #include <list>
 
+#include  <getopt.h>
+
 /**** forward declaration of the calculation functions ****/
 
-/**
- * calculate the force for all particles
- */
+//! function to calculate and update the velocity of the particles
 void calculateF();
 
-/**
- * calculate the position for all particles
- */
+//! function to calculate and update the position of the particles
 void calculateX();
 
-/**
- * calculate the position for all particles
- */
+//! function to calculate and update the velocity of the particles
 void calculateV();
 
-/**
- * plot the particles to a xyz-file
- */
-void plotParticles(int iteration);
 
+//! function to plot the particles using the output writer
+/*!
+\param iteration int representing the number of the current iteration
+*/
+void plotParticles(int iteration);
+//! constant double representing the start time of the simulation.
 constexpr double start_time = 0;
-constexpr double end_time = 1000;
-constexpr double delta_t = 0.014;
+//! double representing the end time of the simulation. Default:1000
+double end_time = 1000;
+//! double representing the timestep of the simulation. Default:0.014
+double delta_t = 0.014;
+//! ParticleContainer containing all particles in the simulation
+ParticleContainer particles;
+
+//! int representation of the output writer being used
+/*!
+*writer_flag=0 -> vtk writer
+*writer_flag=1 -> xyz  writer
+*/
+int writer_flag=0;
+//! main function of the Molecular Simulation
+int main(int argc, char *argsv[]) {
 
-// TODO: what data structure to pick?
-std::list<Particle> particles;
+  // handle options and arguments passed to the executable using getopt
+  int help_flag = 0;
+  char *input_file = nullptr;
 
-int main(int argc, char *argsv[]) {
+  option long_options[] = {
+    {"help", no_argument, &help_flag, 1},
```

Comment:
It's `C-stye` (and the previously mentioned non-portable point), but I very much like that you (re-)use here existing functionality 👍